### PR TITLE
ENT-4271: Added ADR for Braze User Creation

### PR DESCRIPTION
**Pull request by irfanuddinahmad**
**Merge checklist:**
- [x] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production
    - `test.in` for test requirements
    - `make upgrade && make requirements` have been run to regenerate requirements
- [x] [Version](https://github.com/edx/ecommerce-worker/blob/master/setup.py) bumped

**Post merge:**
- [x] Tag pushed and a new [version](https://github.com/edx/ecommerce-worker/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [x] After versioned build finishes in [Travis](https://travis-ci.org/github/edx/ecommerce-worker), verify version has been pushed to [PyPI](https://pypi.org/project/edx-ecommerce-worker/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [ecommerce](https://github.com/edx/ecommerce) to upgrade dependencies (including ecommerce-worker)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in ecommerce will look for the latest version in PyPi.
    - Note: the ecommerce-worker constraint in ecommerce **must** also be bumped to the latest version in PyPi.
- [ ] Deploy `ecommerce`
- [ ] Deploy `ecomworker` on GoCD.
    - While some functions in ecommerce-worker are run via ecommerce, others are handled by a standalone AMI and must be
      released via GoCD.

### CODE REVIEW

File: docs/decisions/0001-braze-user-creation.rst [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+#. Since the potential learners donot have an lms_user_id, the external_user_id can be temporarily mapped to their email address with an Enterprise prefix.
+e.g. 'external_user_id': 'Enterprise-someemail@someemail.com'
+
+#. A batch job would periodically remove the duplicate accounts in Braze for the potential learners once they graduate to learner status in edX
+
+Consequences
+============
+
+#. Potential users can be sent emails via the Braze system. Sailthru can be discontinued.
```
Comment (moconnell1453) [inline docs/decisions/0001-braze-user-creation.rst:36]: Should we also note that we could potentially have email addresses in Braze that never graduate to learner status? Do we have an obligation to remove those after a period of time?

**Review by moconnell1453 — APPROVED** [review]

File: docs/decisions/0001-braze-user-creation.rst (hunk 1 above)
Comment (irfanuddinahmad) [reply docs/decisions/0001-braze-user-creation.rst:36]: Added

File: docs/decisions/0001-braze-user-creation.rst [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+Status
+======
+
+Draft
+
+Context
+=======
+
+EdX Enterprise theme has been using Sailthru for emails to learners and potential learners. Since the potential learners donot yet have an account on the edX platform, their account on Sailthru is created when an email is initiated to their email address. This account creation is automatically performed by Sailthru.
```
Comment (saleem-latif) [inline docs/decisions/0001-braze-user-creation.rst]: nit: replace `donot` with `do not`

File: docs/decisions/0001-braze-user-creation.rst [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+With transition to Braze as the multi-channel communication and analytics platform, the account creation for the potential learners would be initiated from the edX platform. This is necessitated by the user onboarding flow in Braze.
+
+The Braze configuration for edX has a mapping for the lms_user_id field in LMS to Braze's external_user_id field. In case of unavailability of the external_user_id, Braze also has a user alias object that can be used to identify a user. Each user alias consists of two parts: a label, which defines the key of the alias, and a name, which defines the value. An alias name for any single label must be unique across the user base. More details at https://www.braze.com/docs/user_guide/data_and_analytics/user_data_collection/user_profile_lifecycle/
```
Comment (saleem-latif) [inline docs/decisions/0001-braze-user-creation.rst]: mark variable names by back ticks. i.e. \`lms_user_id\`

Comment (saleem-latif) [reply docs/decisions/0001-braze-user-creation.rst]: also \`external_user_id\`

File: docs/decisions/0001-braze-user-creation.rst [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+=========
+
+#. For potential enterprise learners, if an alias is to be created, the label would hold the value 'Enterprise' and the key would be their email address.
+e.g.
+  "user_alias" : {
+      "alias_name" : "Enterprise",
+      "alias_label" : "someuser@someorg.org"
+  }
+
+#. Since the potential learners donot have an lms_user_id, the external_user_id can be temporarily mapped to their email address with an Enterprise prefix.
```
Comment (saleem-latif) [inline docs/decisions/0001-braze-user-creation.rst]: nit:  replace `donot` with `do not`

Comment (saleem-latif) [reply docs/decisions/0001-braze-user-creation.rst]: mark variable names by back ticks. i.e. \`lms_user_id\` and \`external_user_id\`

**Review by saleem-latif — APPROVED** [review]
a few nits, overall Looks Good 👍

**Pull request merged**